app/search_builders/hyrax/my/collections_search_builder.rb

```diff
@@ -21,6 +21,6 @@ def show_only_collections_deposited_by_current_user(solr_parameters)
   # This overrides the models in FilterByType
   # @return [Array<Class>] a list of classes to include
   def models
-    [::AdminSet, Hyrax::AdministrativeSet, Hyrax.config.collection_model.safe_constantize].compact
+    [::AdminSet, Hyrax::AdministrativeSet, ::Collection, Hyrax.config.collection_class].uniq.compact
```

Comment:
This is the fix that allows all the user's collections to appear in the Dashboard -> Collections list.